pkg/controller/endpointslicemirroring/reconciler.go

```diff
@@ -196,9 +197,15 @@ func (r *reconciler) reconcileByPortMapping(
 		// if >0 existing slices, mark all but 1 for deletion.
 		slices.toDelete = existingSlices[1:]
 
-		// Return early if first slice matches desired endpoints.
+		// generated slices must mirror all endpoints annotations but EndpointsLastChangeTriggerTime
+		compareAnnotations := cloneAndRemoveKey(endpoints.Annotations, corev1.EndpointsLastChangeTriggerTime)
+		compareLabelsTemp := cloneAndRemoveKey(existingSlices[0].Labels, discovery.LabelManagedBy)
+		compareLabels := cloneAndRemoveKey(compareLabelsTemp, discovery.LabelServiceName)
```

Comment:
Tiny nit: This feels like we're wasting some effort. We're iterating through labels to clone 2x and then again for  DeepEqual below. It seems like this could ideally be handled in a single loop,  maybe with a `sets.String` with excluded labels as a param.

Reply:
yeah, this is very ugly ... I've modified `cloneAndRemoveKeys`  to be variadic, so we only iterate once. I've compared with the sets.String inplementation and I think that this function does exactly the same operations